src/problems/towers.clj

```diff
@@ -0,0 +1,107 @@
+(ns problems.towers)
+
+(defrecord State [vec-of-pegs])
+
+(def start-state (->State [[5 4 3 2 1] [] [] [] []]))
+(def goal-state (->State [[] [] [] [] [5 4 3 2 1]]))
+
+;;
+;; Determines whether or not a given state is valid
+;; Checks that no disks are out of order
+;; Checks that no disks are duplicated
+;;
+;; Returns a boolean
+;;
+(defn legal? [new-state]
+  (let [stack-order (and (map #(if (< (count %) 2)
+                                        true
+                                        (apply > %)) (:vec-of-pegs new-state)))
+
+        all-disks (apply concat (:vec-of-pegs new-state))
+        correct-disks (= (sort all-disks) (range 1 (+ 1 (count all-disks))))
+        ]
+    (and (reduce #(and %1 %2) stack-order) correct-disks)))
+
+;;;;;;; Helper Functions for Fitness ;;;;;;;
+(defn abs [n] (max n (- n)))
+
+
+(defn fitness-diff [goal-state current-state]
+	(reduce + (map #(abs(- (count %2) (count %1))) (:vec-of-pegs goal-state) (:vec-of-pegs current-state)))
+)
+
+(defn fitness-lonely-disk [state]
+	(count (filter #(= (count %) 1) (:vec-of-pegs state)))
+)
+
+
+;;
+;; Determines the fitness of a state.
+;; Takes in a state, wich we assume to be valid.
+;; Returns an integer... ~~High numbers are "better"~~
+;; According to Nic, higher numbers are worse...
```

Comment:
These two lines of comment are weird, and don't clarify what exactly is the state in your code. I _think_ your `fitness` function now has low numbers as good. You're certainly minimizing the differences between the current state and the goal state, as well as minimizing  the number of pegs with exactly one disk. You have other "points" in the comments below (lines 44-47), but it seems like a lot of them aren't actually used (anymore). It would have been good to clean this up.